### ISSUE TITLE
Pair wall plug mini as non secure

### ISSUE THREAD

**Issue by kungktd** [yhaugom/com.telldus#24]
Can you please add code to add the [wall plug mini](https://telldus.com/produkt/plugin-receiver-mini-z-wave/) as non secure?

Athom’s latest Fibaro app has example code for how to do this in Homey.
Got this response from Athom support...

> You can refer them to the Fibaro sensor implementation in the latest app update if they need an example how to.

**Comment by yhaugom** [comment]
If you could branch the repo and make a pull request with the implementation it would be great. Don't have much spare time at the moment.

**Comment by kungktd** [comment]
I would if I could but I have no clue how to do anything more then use apps from skilled people like you. :)

**Comment by kungktd** [comment]
I have just added `"includeSecure": false,` at line 481 in  app.json and installed the app via CLI. It works fine and my two Mini plugs are now added as non secure.
I have no idea how to branch and make a pull rquest thoe, sorry. But if/when you get the time, pleae consider include the change.

**Comment by yhaugom** [comment]
@kungktd If that is the only thing required, I can fix it tomorrow probably. Thanks for looking into it.

**Comment by yhaugom** [comment]
> I have just added `"includeSecure": false,` at line 481 in app.json and installed the app via CLI. It works fine and my two Mini plugs are now added as non secure.
If I do it like this, then I am thinking the option to add as secure will go away? Can you confirm this @kungktd ?
I didn’t find the time today unfortunately. Sorry, I will try again during this coming week. It might be some changes required to be compatible with the latest SDK too. I am not sure yet.

**Comment by kungktd** [comment]
I think so. Is it possible to add an option in the setting of the app if you want to add Devi we as secure or non secure?

I talked to Telldus and they told me that only doors, locks, garage doors, alarms etc. should be added secure. Wall plugs, lights, switches etc. should be added as non secure.
The Z-wave protocol is already encrypted and adding devices that doesn’t need to be secure as secure makes stress your Z-wave.
That was the explanation I got from Telldus.

**Comment by yhaugom** [comment]
I have tried to confirm this statement, but it is hard with little time to find the information needed. I would need a reference to the z-wave standard explaining this. I've read that Z-wave is not encrypted as standard, and I do not agree that I should make the decision for the user whether they shall include the device as secure or non-secure. If I buy a device that supports secure inclusion, I would expect it to be able to include as a secure device.
The statement from Telldus that only doors, locks, garage doors and alarms should be included secure is not one I agree to if it is not encrypted.
I do not know what units people are connecting to their plugs. It could be a siren for example , or another device they want to be switched on/off securely. It is not possible to know every scenario.
This is why I am reluctant to add the flag. But if it is possible to have this as an option for the user, or someone can point me in the direction of an article confirming that Z-wave is encrypted by default, I will consider it again.
[This blog-post was the first hit when I googled "Is Z-wave encrypted?",](https://www.protectamerica.com/home-security-blog/tech-tips/is-z-wave-encrypted-a-quick-tech-guide_21868#targetText=But%20it's%20true%20%E2%80%94%20Z%2DWave,cause%20other%20types%20of%20damage.) but I know one shouldn’t take blog-post for facts.  I would need a reliable source.
